src/transformers/models/llama/tokenization_llama.py

```diff
@@ -187,15 +209,10 @@ def _tokenize(self, text):
         the extra `SPIECE_UNDERLINE` prepended.
         """
         if not self.legacy:
-            is_first = text.startswith(SPIECE_UNDERLINE)
-            if is_first:
-                text = text[1:]
-
-        tokens = self.sp_model.encode(text, out_type=str)
-
-        if not self.legacy and not is_first and not text.startswith(" ") and tokens[0].startswith(SPIECE_UNDERLINE):
-            tokens = ([tokens[0][1:]] if len(tokens[0]) > 1 else []) + tokens[1:]
-        return tokens
+            text = self.unk_token + text
+            tokens = self.sp_model.encode(text, out_type=str)
+            return tokens[self.unk_token_length :]
+        return self.sp_model.encode(text, out_type=str)
```

Comment:
nit: as the non-legacy code is the 'main' code now, it would be cleaner to have an early return in the case of `self.legacy=True` and the main logic unindented

```suggestion
        if self.legacy:
            return self.sp_model.encode(text, out_type=str) 
        text = self.unk_token + text
        tokens = self.sp_model.encode(text, out_type=str)
        return tokens[self.unk_token_length :]
```